config/auth_permutations_test.go

```diff
@@ -473,3 +497,62 @@ func TestConfig_AuthTypeFromEnv(t *testing.T) {
 		AssertHost: "https://x",
 	}.apply(t)
 }
+
+func TestConfig_MetadataServerInvalidAuthServerUrl(t *testing.T) {
+	configFixture{
+		AuthType:           "metadata-service",
+		MetadataServiceURL: "not-valid-hostname",
+		Host:               "x",
+		AssertError:        `default auth: metadata-service: invalid auth server URL: not-valid-hostname. Config: host=https://x, metadata_service_url=***`,
+	}.apply(t)
+}
+
+func TestConfig_MetadataServerInvalidAuthServerUrl2(t *testing.T) {
+	configFixture{
+		AuthType:           "metadata-service",
+		MetadataServiceURL: "https://google.com",
+		Host:               "x",
+		AssertError:        `default auth: metadata-service: invalid auth server URL: https://google.com. Config: host=https://x, metadata_service_url=***`,
+	}.apply(t)
+}
+
+func TestConfig_MetadataServerConfigure(t *testing.T) {
+	configFixture{
+		AuthType:           "metadata-service",
+		MetadataServiceURL: "http://localhost/token",
+		Host:               "x",
+		AssertHost:         "https://x",
+		AssertAuth:         "metadata-service",
+		AssertHeaders: map[string][]string{
+			"Authorization": {"Bearer XXX-2"}, // XXX-1 is returned during Configure()
+		},
+		HTTPFixtures: []qa.HTTPFixture{
+			{
+				Method:   "GET",
+				Resource: "/token",
+				ExpectedHeaders: map[string][]string{
+					"X-Databricks-Metadata-Version": {"1"},
+					"X-Databricks-Host":             {"https://x"},
+				},
+				Response: config.AzureMsiToken{
+					AccessToken: "XXX-1",
+					ExpiresOn:   json.Number(strconv.FormatInt(time.Now().Add(1*time.Second).Unix(), 10)),
```

Comment:
this is not really portable across languages with `deco code authtests`. I think we need to add a template variable somewhere

Reply:
Good point. I believe the actual time is not necessary for these tests, will update these.